### PR TITLE
Ios 18

### PR DESCRIPTION
**Pull request by dekeract10**
Corrected bugs which were detected in testing.

### CODE REVIEW

File: Receipt Wallet/CreateWarrantyView.swift [hunk 1]
```diff
@@ -297,6 +291,24 @@ class CreateWarrantyView: UIView {
         gradient.frame = formScrollView.frame
     }
     
+    func addViewIdentifiers(){
+        nameInputLabel.accessibilityIdentifier = "nameInputLabel"
```
Comment (MilePrastalo) [inline Receipt Wallet/CreateWarrantyView.swift]: accessibility identifier should set at view definition inside lazy var...

File: Receipt Wallet/SelectCategoryView.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            return categoryButton
+        }
+        return buttons
+    }()
+    
+    func setupView(){
+        backgroundColor = .green
+        addSubview(menuLabel)
+        addSubview(scrollView)
+        NSLayoutConstraint.activate([
```
Comment (MilePrastalo) [inline Receipt Wallet/SelectCategoryView.swift]: stick to one format for setting constraints

File: Receipt Wallet/SelectCategoryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @objc func categoryButtonTapped(sender: UIButton) {
+        if let catButton = sender as? CategoryButton, let selectedCategory = catButton.category {
+            self.delegate?.categoryDidSelect(category: selectedCategory)
+        }
+    }
+    
+    override func loadView() {
+        super.loadView()
+        self.view = selectCategoryView
+        selectCategoryView.buttons.forEach{button in
```
Comment (MilePrastalo) [inline Receipt Wallet/SelectCategoryViewController.swift]: adding target and filtering should be moved to viewDidLoad

**Review by MilePrastalo — CHANGES_REQUESTED** [review]
see comments

**Review by mknzvc — APPROVED** [review]

**Review by MilePrastalo — APPROVED** [review]

**Pull request merged**